valeriano-manassero/trino/templates/deployment-coordinator.yaml

```
@@ -88,10 +96,65 @@ spec:
           configMap:
             name: {{ .configMapName }}
         {{- end }}
-      {{- if .Values.config.coordinator.initContainers }}
+      {{- if or .Values.config.coordinator.initContainers (include "trino.tlsEncryption" . ) }}
       initContainers:
-      {{-  tpl (toYaml .Values.config.coordinator.initContainers) . | nindent 6 }}
+      {{- if (include "trino.tlsEncryption" . ) }}
+        - name: init-certs
+          image: {{ .Values.initKeystore.image.repository }}:{{ .Values.initKeystore.image.tag }}
+          imagePullPolicy: {{ .Values.initKeystore.image.pullPolicy }}
+          {{- with .Values.image.securityContext }}
+          securityContext:
+            runAsUser: {{ .runAsUser }}
+            runAsGroup: {{ .runAsGroup }}
+          {{- end }}
+          command: [ /bin/bash ]
+          args:
+          - -ec
+          - |-
+            {{- if .Values.tls.autoGenerated }}
+            if [[ -f "/certs/tls.key" ]] && [[ -f "/certs/tls.crt" ]]; then
+              openssl pkcs12 -export -in "/certs/tls.crt" \
+                  -passout pass:"${TRINO_KEYSTORE_PASSWORD}" \
+                  -inkey "/certs/tls.key" \
+                  -out "/tmp/keystore.pfx"
+              keytool -importkeystore -srckeystore "/tmp/keystore.pfx" \
+                  -srcstoretype PKCS12 \
+                  -srcstorepass "${TRINO_KEYSTORE_PASSWORD}" \
+                  -deststorepass "${TRINO_KEYSTORE_PASSWORD}" \
+                  -destkeystore "{{ .Values.config.general.path }}/certs/keystore"
+              rm "/tmp/keystore.pfx"
+            else
+                echo "Couldn't find the expected PEM certificates! They are mandatory when encryption via TLS is enabled."
+                exit 1
+            fi
+            {{- else }}
+            if [[ -f "/certs/keystore" ]]; then
+                cp "/certs/keystore" "{{ .Values.config.general.path }}/certs/keystore"
+            else
+                echo "Couldn't find the expected Java Key Stores (JKS) files! They are mandatory when encryption via TLS is enabled."
+                exit 1
+            fi
+            {{- end }}
+          env:
+            {{- if .Values.tls.keystorePasswordSecret }}
+            - name: TRINO_KEYSTORE_PASSWORD
+              valueFrom:
+                secretKeyRef:
+                  name: {{ .Values.tls.keystorePasswordSecret }}
+                  key: keystore-password
+            {{- end }}
+          {{- if .Values.tls.resources }}
+          resources: {{- toYaml .Values.tls.resources | nindent 10 }}
+          {{- end }}
+          volumeMounts:
+          - name: certs
+            mountPath: /certs
+          - name: certs-shared
+            mountPath: {{ .Values.config.general.path }}/certs
       {{- end }}
+      {{- if .Values.config.coordinator.initContainers }}
+      {{-  tpl (toYaml .Values.config.coordinator.initContainers) . | nindent 8 }}
+      {{- end }}{{- end }}
```

I find this line confusing. I _think_ it's closing the `if` blocks from lines 99 and 101, but since those are on separate lines, I would put the `end` statements on separate lines as well.
```suggestion
      {{- end }}
      {{- end }}
```